### DISCUSSION TITLE
Shallow import issue on React Native

### DISCUSSION THREAD

**Discussion by glitch-txs** [pmndrs/zustand#2127] [Bug report] [answered]
## Summary

Hi, while installing zustand in a React Native app and calling shallow it will throw a warn even when it's NOT imported as default.

```ts
import { shallow } from 'zustand/shallow'

shallow()
```

```bash
WARN  [DEPRECATED] Default export is deprecated. Instead use `import { shallow } from 'zustand/shallow'`.
```

This doesn't happen in zustand 4.3.1

the issue seems to be in the way the code was compiled.

Check on [npm the last lines](https://www.npmjs.com/package/zustand?activeTab=code) in `shallow.js`

```ts
exports.default = shallow;
exports.shallow = shallow$1;

module.exports = shallow$1;
module.exports.shallow = shallow;
exports.default = module.exports;
```

by changing it for this it works well

```ts
exports.default = shallow;
exports.shallow = shallow$1;

module.exports = shallow$1;
module.exports.shallow = shallow$1; // <-- change here
exports.default = module.exports;
```

## Link to reproduction

https://github.com/glitch-txs/zustand-rn-warn/blob/main/App.tsx

**Comment by dbritto-dev** [comment]
@glitch-txs weird, there's no warnings -> https://codesandbox.io/p/sandbox/magical-star-f7r88x?file=%2Fsrc%2FApp.tsx%3A7%2C1

**Reply by glitch-txs** [reply]
Hi, somehow this only happens in React Native but it works well in React + Vite. I tested both

**Reply by dbritto-dev** [reply]
Maybe something related to Metro bundler I guess

**Comment by dai-shi** [comment]
Oops, look like some recent change broke our manual patching...

**Reply by dai-shi** [reply, marked as answer]
#2128 should fix it.

https://ci.codesandbox.io/status/pmndrs/zustand/pr/2128
Please try "Local Install Instructions" ☝️

**Reply by glitch-txs** [reply]
Thanks! This works great! 🤝
